### ISSUE TITLE
Support boolean conversion to uin8

### ISSUE THREAD

**Issue by mfridman** [ClickHouse/clickhouse-go#621]
Going from driver v1 to v2, there is an issue with saving booleans.

E.g., in v1 I could create a table `is_applied UInt8` and insert a `boolean` (presumably a conversion happened under the hood).

But with v2 I'm noticing this is no longer the case?

```
unexpected error: code: 47, message: Missing columns: 'true' while processing query: 'true', required columns: 'true' 'true'
```

It'd be ideal if there was a type conversion for bool <-> uint8.

**Comment by gingerwizard** [comment]
@mfridman yes there are some gaps on time conversions defn. We resolve as people raise them at the moment. The previous driver also allowed conversions which implicitly could result in a loss of precision. We don't allow these now.

boolean to uint8 seems reasonable however, but uint8 to boolean we wont do - for precision loss reasons.

So we can support:

1. Reading a boolean column into a uint8
2. Inserting a boolean into a uint8 column

Do you need anything else?

**Comment by mfridman** [comment]
Just the boolean conversion.

For context, we're trying to improve ClickHouse support in a migration tool, and users might already have the table and field  `is_applied` as `UInt8`

```
CREATE TABLE goose_db_version (
...
  is_applied UInt8,
...
) Engine = MergeTree(date, (date), 8192)
```

But most of the application-level logic assumes we can insert a `boolean` and it'll get converted to a number. 

Although there is an argument we should handle this in the application not in the driver. So I'd also understand if you want to keep things explicit.
